Feature/subjects

brunosra opened the pull request.

Commit:
added subjects models and migrations

Commit:
renamed tables

Commit:
added user and subjects

Commit:
added more models

Commit:
renamed subjects to subject

Commit:
added 2 more classes

Commit:
fixed Integer error

Commit:
removed plurals from tables

Commit:
removed plurals

Commit:
added gender relationship to users

Commit:
changed genders to singular

Commit:
added null to gender_other

Commit:
fixed classs to students_class

Commit:
added school_year relationships

Commit:
cleaned up code

Commit:
added relationship tables

Commit:
added enums

albertoturrion commented on cocciep/models/student.py:
```diff
@@ -0,0 +1,17 @@
+from cocciep.db import db
+from datetime import datetime
+
+class Student(db.Model):
```
Wouldn't be _Students_ model?

albertoturrion reviewed (commented):
I have seen that in many-to-many relationships Flask documentation strongly recommends doesn't create helper table as a model, but an actual table https://flask-sqlalchemy.palletsprojects.com/en/2.x/models/

Shouldn't we create a table for StudentsClass?

albertoturrion committed:
Update user.py

Changing user to teacher

albertoturrion committed:
Changing user to teacher

albertoturrion committed:
Changing user to teacher and adding students to gender

I consider that students should be here too, Student model also has gender property

albertoturrion committed:
Adding relationships to Subject model

As ClassesSubjectsTeachers is a helper table, we need to make the relationship between the 3 models (many-to-many)
https://flask-sqlalchemy.palletsprojects.com/en/2.x/models/

albertoturrion committed:
Daily relationship to enumStatusClass

Creating the relationship between Daily and EnumStatusClass. It's been created a new property "status" within Daily model

albertoturrion committed:
New proposal for Enums

I have read that we need to create as the second way: https://stackoverflow.com/questions/36136112/defining-sqlalchemy-enum-column-with-python-enum-raises-valueerror-not-a-valid

brunosra committed:
Merge pull request #3 from brunosra/EnumGender

Changing user to teacher and adding students to gender

brunosra committed:
Merge pull request #4 from brunosra/fix/SubjectRelationships

Adding relationships to Subject model

albertoturrion committed:
Merge pull request #5 from brunosra/fix/statusRelationship

Daily relationship to enumStatusClass
